document/core/util/mathdef.py

```diff
@@ -39,13 +38,36 @@ def latex_hyperlink(file, id):
   return '\\hyperref[%s:%s]' % (file, id)
 
 def latex_transform_math_xref(node):
-  node['latex'] = \
-    xref_re.sub(lambda m: latex_hyperlink(m.group(1), m.group(2)), node['latex'])
+  new_text = xref_re.sub(lambda m: latex_hyperlink(m.group(1), m.group(2)), node.astext())
+  node.children[0] = nodes.Text(new_text)
+
+def latex_visit_math(self, node):
```

Comment:
It seems strange to me that we'd have to copy the code here; is it possible that there's a fix that can be made upstream so we could use their versions of this code?

Reply:
Perhaps. But until we catch up with upstream, we won't be able to take any fix they make anyway :)
I think the root problem is how we are using `add_node` below to override behavior, `add_node` with `override=True` basically means you have to duplicate all the work, because the method is overwritten.

In theory, we could store the method before it gets overwritten, `m = translator.visit_math`, but the problem we `translator` does not exist, until `latex_visit_math` is called, which by the time, `visit_math` has already been overwritten.

Reply:
Can we flag the duplicated code with TODOs for now?

Reply:
Good idea, added a TODO there. I'll also go dig around Sphinx to see if they have a recommended way to extend nodes.